### PR TITLE
Untrained exception

### LINKED ISSUE #21: TypeError: 'NoneType' object has no attribute '__getitem__'
https://github.com/mozilla/spicedham/blob/master/spicedham/bayes.py#L42

`total` there can be None since `get_key` returns None by default of the thing isn't in the store. If `total` is None, then this kicks up a `TypeError`.

We should either fix this to handle when `total = None` or change the `get_key` call to return a more appropriate default that allows this code to work.

### PR DESCRIPTION
**Pull request by iankronquist**
When the classifier isn't trained you shouldn't be able to use it.
Even in tests.
Fix #21

### CODE REVIEW

File: spicedham/bayes.py [hunk 1]
```diff
@@ -37,9 +38,12 @@ def train(self, result, is_spam):
 
     def classify(self, response):
         """
-        Get the probability that a response is spam. response is a list
+        Get the probability that a response is spam. response is a list.
+        Raise value error if the classifier has not yet been trained.
         """
         total = self.backend.get_key(self.__class__.__name__, '*')
+        if total == None:
+            raise ValueError("The classifier has not yet been trained.")
```
Comment (willkg) [inline spicedham/bayes.py]: Nit: This isn't really what ValueError is for. This should be a new exception that's spicedham specific. Maybe an ImproperlyConfigured or something like that.

Comment (iankronquist) [inline spicedham/bayes.py]: I'll go ahead and do that, but both Lonnen and @mythmon recommended I use a built in exception.

**Comment by willkg** [comment]
I just talked with @mythmon. They're totally right in that it's generally better to use built-in exceptions when appropriate. I'm not sure this is a good case for ValueError, though.

https://docs.python.org/2/library/exceptions.html#exceptions.ValueError

I'm not sure I see a good built-in exception for this. Plus I'm thinking about how this will affect fjord and I think I'd feel better about handling (and ignoring) an ImproperlyConfigured error that I know came from Spicedham and not something Spicedham was using than something like ValueError where I'm not entirely sure it came from Spicedham.

**Comment by willkg** [comment]
Having said that, I'm not hung up on this. Either way is fine. I'm just thinking out loud. Plus I feel really bad about disagreeing with @mythmon and @lonnen. I'll plan to do penance later.

**Comment by iankronquist** [comment]
@willkg Alright, let's try this again. Sorry about the flip-flopping there.

**Comment by willkg** [comment]
No worries about "flip-flopping"! This looks good to me! r+

**Pull request merged**